perses/tests/test_CUDA.py

```diff
@@ -0,0 +1,213 @@
+                    getPositions=True, enforcePeriodicBox=False
+                ).getPositions(asNumpy=True)
+                old_pos = np.asarray(htf.old_positions(pos))
+                new_pos = np.asarray(htf.new_positions(pos))
+                reverse_eq_new.append(new_pos)
+                reverse_eq_old.append(old_pos)
+
+        # Reverse work (1 -> 0)
+        reverse_works = [integrator.get_protocol_work(dimensionless=True)]
+        for rev_step in range(nsteps_neq):
+            initial_time = time.time()
+            integrator.step(1)
+            elapsed_time = (time.time() - initial_time) * unit.seconds
+            reverse_works.append(integrator.get_protocol_work(dimensionless=True))
+            if rev_step % save_freq_neq == 0:
+                _logger.info(
+                    f"reverse NEQ step: {rev_step}, took: {elapsed_time / unit.seconds} seconds"
+                )
+                pos = context.getState(
+                    getPositions=True, enforcePeriodicBox=False
+                ).getPositions(asNumpy=True)
+                old_pos = np.asarray(htf.old_positions(pos))
+                new_pos = np.asarray(htf.new_positions(pos))
+                reverse_neq_old.append(old_pos)
+                reverse_neq_new.append(new_pos)
+        reverse_works_master.append(reverse_works)
+
+        # Save output
+        # create output directory if it does not exist
+        out_path = pathlib.Path(outdir_path)
+        out_path.mkdir(parents=True, exist_ok=True)
+        # Save works
+        with open(os.path.join(out_path, f"forward.npy"), "wb") as out_file:
+            np.save(out_file, forward_works_master)
+        with open(os.path.join(out_path, f"reverse.npy"), "wb") as out_file:
+            np.save(out_file, reverse_works_master)
+
+        # Save trajs
+        with open(os.path.join(out_path, f"forward_eq_old.npy"), "wb") as out_file:
+            np.save(out_file, np.array(forward_eq_old))
+        with open(os.path.join(out_path, f"forward_eq_new.npy"), "wb") as out_file:
+            np.save(out_file, np.array(forward_eq_new))
+        with open(os.path.join(out_path, f"reverse_eq_new.npy"), "wb") as out_file:
+            np.save(out_file, np.array(reverse_eq_new))
+        with open(os.path.join(out_path, f"reverse_eq_old.npy"), "wb") as out_file:
+            np.save(out_file, np.array(reverse_eq_old))
+        with open(os.path.join(out_path, f"forward_neq_old.npy"), "wb") as out_file:
+            np.save(out_file, np.array(forward_neq_old))
+        with open(os.path.join(out_path, f"forward_neq_new.npy"), "wb") as out_file:
+            np.save(out_file, np.array(forward_neq_new))
+        with open(os.path.join(out_path, f"reverse_neq_old.npy"), "wb") as out_file:
+            np.save(out_file, np.array(reverse_neq_old))
+        with open(os.path.join(out_path, f"reverse_neq_new.npy"), "wb") as out_file:
+            np.save(out_file, np.array(reverse_neq_new))
```

Comment:
@mikemhenry As far as I see this seems very similar (if not the same) as the example we have already for protein neq switching. Other than having to specify the CUDA platform, are there any reasons why we need this in a separate test/file?

I can think of rewriting the current examples to use the "best" platform available and it should use CUDA when available, I think.

Reply:
Tested this with my local setup and it runs in just a few mins (less than 10min for all of the examples). So, is it crazy to make all of these run in the GPU CI workflow? We might just use something like `configure_platform(utils.get_fastest_platform().getName())` for the platform in the examples scripts, and my understanding is that they should use CUDA when available. What do you think?

Reply:
I'm afraid that if we don't explicitly specify `CUDA` then we won't know if the GPU stuff stops working and falls back to CPU. So maybe we just have a quick test where we specifically ask for CUDA, and for the rest we just ask for the fastest?

Reply:
Can we also run one test (doesn't have to be all) on `OpenCL` as well?